recon_surf/recon-surfreg.sh

```diff
@@ -0,0 +1,499 @@
+for hemi in rh lh; do
+
+  # Check if running on an existing registry
+  if [ -f "$SUBJECTS_DIR/$subject/surf/$hemi.sphere" ] || [ -f "$SUBJECTS_DIR/$subject/surf/$hemi.sphere.reg" ] || [ -f "$SUBJECTS_DIR/$subject/surf/$hemi.angles.txt" ]; then
+    echo "ERROR: running on top of an existing registry!"
+    echo "The output directory must not contain a surfreg from a previous invocation of recon-surf."
+    exit 1
+  fi
+
+done
+
+# collect info
+StartTime=`date`;
+tSecStart=`date '+%s'`;
+year=`date +%Y`
+month=`date +%m`
+day=`date +%d`
+hour=`date +%H`
+min=`date +%M`
+
+
+sdir=$SUBJECTS_DIR/$subject/surf
+ldir=$SUBJECTS_DIR/$subject/label
+
+
+# Set up log file
+DoneFile=$SUBJECTS_DIR/$subject/scripts/recon-surfreg.done
+if [ $DoneFile != /dev/null ] ; then  rm -f $DoneFile ; fi
+LF=$SUBJECTS_DIR/$subject/scripts/recon-surfreg.log
+if [ $LF != /dev/null ] ; then  rm -f $LF ; fi
+echo "Log file for recon-surfreg.sh" >> $LF
+date  |& tee -a $LF
+echo "" |& tee -a $LF
+echo "export SUBJECTS_DIR=$SUBJECTS_DIR" |& tee -a $LF
+echo "cd `pwd`"  |& tee -a $LF
+echo $0 ${inputargs[*]} |& tee -a $LF
+echo "" |& tee -a $LF
+cat $FREESURFER_HOME/build-stamp.txt |& tee -a $LF
+echo $VERSION |& tee -a $LF
+uname -a  |& tee -a $LF
+
+
+# Print parallelization parameters
+echo " " |& tee -a $LF
+if [ "$DoParallel" == "1" ]
+then
+  echo " RUNNING both hemis in PARALLEL " |& tee -a $LF
+else
+  echo " RUNNING both hemis SEQUENTIALLY " |& tee -a $LF
+fi
+echo " RUNNING $OMP_NUM_THREADS number of OMP THREADS " |& tee -a $LF
+echo " RUNNING $ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS number of ITK THREADS " |& tee -a $LF
+echo " " |& tee -a $LF
+
+
+#if false; then
+
+########################################## START ########################################################
+
+
+# ================================================== SURFACES ==========================================================
+
+# look into the recon-surf.log to find the voxel sizes
+vox_size=`cat $SUBJECTS_DIR/$subject/scripts/recon-surf.log | grep -E " - Voxel Size " | cut -d' ' -f5 | cut -d'x' -f1 | sed '1p;d'`
+
+
+if (( $(echo "$vox_size < $hires_voxsize_threshold" | bc -l) ))
+then
+  echo "The voxel size $vox_size is less than $hires_voxsize_threshold, so we are proceeding with hires options." |& tee -a $LF
+  hiresflag="-hires"
+else
+  echo "The voxel size $vox_size is not less than $hires_voxsize_threshold, so we are proceeding with standard options." |& tee -a $LF
+  hiresflag=""
+fi
+
+CMDFS=""
+
+for hemi in lh rh; do
+
+  CMDF="$SUBJECTS_DIR/$subject/scripts/$hemi.processing.cmdf"
+  CMDFS="$CMDFS $CMDF"
+  rm -rf $CMDF
+
+  echo "echo \" \"" |& tee -a $CMDF
+  echo "echo \"============ Creating surfaces $hemi - FS sphere, surfreg ===============\"" |& tee -a $CMDF
+  echo "echo \" \"" |& tee -a $CMDF
+
+  # Surface registration for cross-subject correspondance (registration to fsaverage)
+    cmd="recon-all -subject $subject -hemi $hemi -sphere $hiresflag -no-isrunning $fsthreads"
```

Comment:
This just calls mris_sphere , see https://github.com/freesurfer/freesurfer/blob/483c25eef64b6352a93e950b9ad31a4d90117e10/scripts/recon-all#L3875 
for that $hiresflag is not needed, so you can remove it here and the block above that checks voxel sizes.

Reply:
The same command is used in the [recon-surf.sh](https://github.com/Deep-MI/FastSurfer/blob/3c84f9d8e36a4836d26a6a47919c9d315b83093b/recon_surf/recon-surf.sh#L825). Since the `$hires` flag is not needed for `mris_sphere`, it can be removed there as well?

Reply:
in recon-surf the highresflag is passed to all recon-all calls because of two reasons: 
- no time to check which individual block really use it, sometimes we call multiple at a time
- to be future proof, in case that high-res functionality is added in the future to a specific block.
So I would, for now, pass it everywhere, since it is populated anyway. 
Here I think we can go either way, since it is a single command we may even want to directly call the mris_sphere command instead of using the recon-all as a wrapper.